language/move-prover/interpreter/crypto/Cargo.toml

```diff
@@ -13,5 +13,17 @@ diem-workspace-hack = { path = "../../../../common/workspace-hack" }
 anyhow = "1.0.38"
 curve25519-dalek = { version = "0.1.0", package = "curve25519-dalek-fiat", default-features = false, features = ["std"] }
 ed25519-dalek = { version = "0.1.0", package = "ed25519-dalek-fiat", default-features = false, features = ["std", "serde"] }
+proptest = { version = "1.0.0", optional = true }
+proptest-derive = { version = "0.3.0", optional = true }
 sha2 = "0.9.3"
 sha3 = "0.9.1"
+x25519-dalek = { version = "0.1.0", package = "x25519-dalek-fiat", default-features = false, features = ["std"] }
+
+[features]
+default = ["fiat"]
+assert-private-keys-not-cloneable = []
+cloneable-private-keys = []
+fuzzing = ["proptest", "proptest-derive", "cloneable-private-keys"]
+fiat = ["curve25519-dalek/fiat_u64_backend", "ed25519-dalek/fiat_u64_backend", "x25519-dalek/fiat_u64_backend"]
+u64 = ["curve25519-dalek/u64_backend", "ed25519-dalek/u64_backend", "x25519-dalek/u64_backend"]
+u32 = ["curve25519-dalek/u32_backend", "ed25519-dalek/u32_backend", "x25519-dalek/u32_backend"]
```

Comment:
It looks like this was just copy pasted from the diem-crypto crate and that a number of these lines aren't even needed.

Specifically all the proptest and x25519 stuff could be dropped + the following features:
* assert-private-keys-not-cloneable
* cloneable-private-keys
* fuzzing

Reply:
@mengxu-fb mind fixing this so that this file is more maintainable?

Reply:
Will do!

Reply:
Thanks!